src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,295 @@
+
+  /**
+   * Set the sticky-header's z-index as 10 in default. Make it as an input
+   * variable to make user be able to customize the zIndex when
+   * the sticky-header's zIndex is not the largest in current page.
+   * Because if the sticky-header's zIndex is not the largest in current page,
+   * it may be sheltered by other element when being stick.
+   */
+  @Input('cdkStickyHeaderZIndex') zIndex: number = 10;
+  @Input('cdkStickyParentRegion') parentRegion: HTMLElement;
+  @Input('cdkStickyScrollableRegion') scrollableRegion: HTMLElement;
+
+  private _onScrollBind: EventListener = this.onScroll.bind(this);
+  private _onResizeBind: EventListener = this.onResize.bind(this);
+  private _onTouchMoveBind: EventListener = this.onTouchMove.bind(this);
+  public isStuck: boolean = false;
+
+  // the element with the 'cdkStickyHeader' tag
+  public element: HTMLElement;
+
+  // the upper container element with the 'cdkStickyRegion' tag
+  public stickyParent: HTMLElement;
+  public upperScrollableContainer: HTMLElement;
+
+  /**
+   * the original css of the sticky element, used to reset the sticky element
+   * when it is being unstuck
+   */
+  public originalCss: any;
+
+  private _containerStart: number;
+  private _scrollFinish: number;
+
+  private _scrollingWidth: number;
+
+  constructor(private _element: ElementRef,
+              public scrollable: Scrollable,
+              @Optional() public parentReg: CdkStickyRegion) {
+    this.element = _element.nativeElement;
+    this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+    this.scrollableRegion = scrollable.getElementRef().nativeElement;
+    if (parentReg != null) {
+      this.parentRegion = parentReg.getElementRef().nativeElement;
+    }
+  }
+
+  ngAfterViewInit(): void {
+
+    if (this.parentRegion != null) {
+      this.stickyParent = this.parentRegion;
+    }else {
+      this.stickyParent = this.element.parentElement;
+    }
+
+    this.originalCss = {
+      zIndex: this.getCssValue(this.element, 'zIndex'),
+      position: this.getCssValue(this.element, 'position'),
+      top: this.getCssValue(this.element, 'top'),
+      right: this.getCssValue(this.element, 'right'),
+      left: this.getCssValue(this.element, 'left'),
+      bottom: this.getCssValue(this.element, 'bottom'),
+      width: this.getCssValue(this.element, 'width'),
+    };
+
+    this.attach();
+
+    this.defineRestrictionsAndStick();
+  }
+
+  ngOnDestroy(): void {
+    this.upperScrollableContainer.removeEventListener('scroll', this._onScrollBind);
+    this.upperScrollableContainer.removeEventListener('resize', this._onResizeBind);
+    this.upperScrollableContainer.removeEventListener('touchmove', this._onTouchMoveBind);
+  }
+
+  attach() {
+    this.upperScrollableContainer.addEventListener('scroll', this._onScrollBind, false);
+    this.upperScrollableContainer.addEventListener('resize', this._onResizeBind, false);
```

Comment:
Because 'this.defineRestrictions();' and ' this.sticker();' functions need to be called when the scrollable container is scrolled. 
I saw that, in 'scrollable.ts', the interface will only emit a scroll event when the scrollable container is scrolled.